### PR TITLE
Added support for Layered Navigation w/out making any configuration changes

### PR DESCRIPTION
**Pull request by erikhansen**
I installed this Lesti_Fpc on a site and upon testing it, I found out that it wasn't working with layered navigation. I eventually came across [the post](https://gordonlesti.com/lestifpc-and-layered-navigation/) explaining how to make it work, but by that time I had figured out that I needed to add each attribute to the "Uri Params" textarea in the system config. This solution was clunky, so I decided to add support for the layered navigation attributes to be automatically added to the "Uri Params" list.

I included a system config field so that a user can disable this functionality, if they so desire.

I also fixed a couple of misspellings.

### CODE REVIEW

**Comment by erikhansen** [comment]
I also updated the default list of "Uri Params" to include "cat", as it isn't an actual attribute, so it needs to be manually specified.

**Comment by Mic2005** [comment]
Hi Erik,

Correct. We also added the attributes also to the uri field.
We have stable attribuut sets so no need to add items automatically.

MiC
Op 25 mrt. 2015 19:51 schreef "Erik Hansen" <notifications@github.com>:

**Comment by GordonLesti** [comment]
hello @erikhansen 
looks nice, thank you very much. Could you add a test?

**Comment by Mic2005** [comment]
Hi Erik,

Does it add the Layered Navigation attributes to "Uri Params"or overwrite
the "Uri Params" feeld complete?

Micha

2015-03-25 19:51 GMT+01:00 Erik Hansen <notifications@github.com>:

File: app/code/community/Lesti/Fpc/Helper/Data.php [hunk 1]
```diff
@@ -125,7 +133,50 @@ protected function _matchUriParam($expression, $param)
...
+                default:
+                    $filterableField = 'is_filterable';
+            }
+            $attributeCollection->addFieldToFilter($filterableField, true);
+
+            foreach ($attributeCollection as $attribute) {
+                $layeredNavigationAttributes[] = $attribute->getAttributeCode();
+            }
+        }
+        return $layeredNavigationAttributes;
```
Comment (GordonLesti) [inline app/code/community/Lesti/Fpc/Helper/Data.php]: @erikhansen can we cache `$layeredNavigationAttributes`?

**Pull request merged**